editor/editor_interface.cpp

```diff
@@ -263,6 +265,77 @@ void EditorInterface::set_current_feature_profile(const String &p_profile_name)
 	EditorFeatureProfileManager::get_singleton()->set_current_profile(p_profile_name, true);
 }
 
+// Editor dialogs.
+
+void EditorInterface::popup_dialog_scene_tree(const Callable &p_callback, const TypedArray<StringName> &p_valid_types) {
+	if (dialog_scene_tree) {
+		dialog_scene_tree->queue_free();
+	}
+	dialog_scene_tree = memnew(SceneTreeDialog);
+	get_base_control()->add_child(dialog_scene_tree);
+
+	Vector<StringName> valid_types;
+	int length = p_valid_types.size();
+	valid_types.resize(length);
+	for (int i = 0; i < length; i++) {
+		valid_types.push_back(p_valid_types[i]);
```

Comment:
```suggestion
		valid_types.write[i] = p_valid_types[i];
```
Same in the other method.